dialogue-annotations-example/src/main/java/com/palantir/myservice/example/CustomRuntimeAwareDeserializer.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * (c) Copyright 2025 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.myservice.example;
+
+import com.palantir.conjure.java.lib.SafeLong;
+import com.palantir.dialogue.ConjureRuntime;
+import com.palantir.dialogue.Deserializer;
+import com.palantir.dialogue.Response;
+import com.palantir.dialogue.TypeMarker;
+import com.palantir.dialogue.annotations.DeserializerFactory;
+import com.palantir.logsafe.Preconditions;
+import com.palantir.logsafe.exceptions.SafeIllegalStateException;
+import java.util.Optional;
+import javax.annotation.Nullable;
+
+public final class CustomRuntimeAwareDeserializer
+        implements DeserializerFactory<MySerializableType>, Deserializer<MySerializableType> {
```

Comment:
I think this still isn't quite right - we shouldn't be instantiating deserializers in the `DeserializerFactory` constructor.

We also should update the test cases in `dialogue-annotations-processor`, since those example are still using hard-coded object mappers instead of using the provided `ConjureRuntime` (this PR adds a constructor with an unused `ConjureRuntime` parameter, which is a bit strange).

I think we want something like this:

https://github.com/palantir/dialogue/commit/4a6e6a1298daefe9a45e0d114a1747148c29c0e6

It would be good to maintain the consistency between the examples in  `dialogue-annotations-processor` tests and `dialogue-annotations-example`.

Reply:
For instance, this allows you to do more interesting things, like deserialize into a generic type while using the provided `ConjureRuntime` deserializer to deserialize the response into an instance of the generic type parameter.

https://github.com/palantir/dialogue/commit/abd0c7c26ee39c088a19fdfbcda7c3efe746e4f7

Reply:
I see. Thanks for putting that together. I've cherry picked these commits.